sdk/react/src/markers/IconMarker/IconMarker.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import * as React from 'react';
+import { Image, TouchableWithoutFeedback, View } from 'react-native';
+import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet';
+
+import styles from './IconMarker.styles';
+
+type Props = {
+  backgroundColor?: ?string,
+  iconUrl: ?string,
+  imageUrl: ?string,
+  onSeek: () => void,
+  style?: ViewStyleProp,
+  touchColor?: ?string,
+};
+
+type State = {
+  isExpanded: boolean,
+};
+
+export default class IconMarker extends React.Component<Props, State> {
+  static defaultProps = {
+    backgroundColor: undefined,
+    style: undefined,
+    touchColor: undefined,
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isExpanded: false,
+    };
+
+    (this: Object).handlePress = this.handlePress.bind(this);
+  }
+
+  handlePress() {
+    const { imageUrl, onSeek } = this.props;
+    const { isExpanded } = this.state;
+
+    // Trigger seek callback if the marker has no image or has been expanded. If there is an image we have to show it
+    // first and only after the second click on that image trigger the callback.
+    if (!imageUrl || isExpanded) {
+      onSeek();
```

Comment:
I think it will be safer to call it from the callback in `setState`. Now it can be called several times

Reply:
Probably you're right. But it's a callback for press handling, I'm not sure that it can be called several times, could you please elaborate on that?

Reply:
`setState` is asynchronous, so if you tap twice state will still be old. You need to pass a function to `setState` instead of an object and call `onSeek` in the callback

Reply:
Yeah, but we have to onSeek before it will be updated.

Reply:
Eventually I got your point, I though you're talking about the second callback in the setState. Please review.